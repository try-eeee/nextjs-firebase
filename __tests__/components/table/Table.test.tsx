import React from "react";
import { render } from "@testing-library/react";
import {
  Table,
  TableCell,
  TableHeader,
  TableRow,
} from "@/components/table/Table";

describe("Table UnitTest", () => {
  test("Snapshot Test Of Table", () => {
    const { asFragment } = render(
      <Table
        tableHeader={
          <TableRow>
            <TableHeader>テストID</TableHeader>
            <TableHeader>テスト日時</TableHeader>
            <TableHeader>テストタイトル</TableHeader>
          </TableRow>
        }
        tableRow={
          <>
            <TableRow>
              <TableCell headerLabel="テストID" width={150}>
                1122
              </TableCell>
              <TableCell headerLabel="テスト日時" width={150}>
                2024/11/07
              </TableCell>
              <TableCell headerLabel="タイトル" width={150}>
                テストタイトル
              </TableCell>
            </TableRow>
          </>
        }
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
