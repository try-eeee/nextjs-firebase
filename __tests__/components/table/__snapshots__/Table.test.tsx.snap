// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table UnitTest Snapshot Test Of Table 1`] = `
<DocumentFragment>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th>
          テストID
        </th>
        <th>
          テスト日時
        </th>
        <th>
          テストタイトル
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          data-label="テストID"
          width="150"
        >
          1122
        </td>
        <td
          data-label="テスト日時"
          width="150"
        >
          2024/11/07
        </td>
        <td
          data-label="タイトル"
          width="150"
        >
          テストタイトル
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
